authzed/api/v1/experimental_service.proto

```diff
@@ -136,3 +184,174 @@ message BulkExportRelationshipsResponse {
   Cursor after_result_cursor  = 1;
   repeated Relationship relationships = 2;
 }
+
+// Reflection types ////////////////////////////////////////////
+
+message ExperimentalReflectSchemaRequest {
+  Consistency consistency = 1;
+ 
+  // optional_filters defines optional filters that are applied in
+  // an OR fashion to the schema, before being returned
+  repeated ExpSchemaFilter optional_filters = 2;
+}
+
+message ExperimentalReflectSchemaResponse {
+  // definitions are the definitions defined in the schema.
+  repeated ExpDefinition definitions = 1;
+
+  // caveats are the caveats defined in the schema.
+  repeated ExpCaveat caveats = 2;
+
+  // read_at is the ZedToken at which the schema was read.
+  ZedToken read_at = 3;
+}
+
+message ExpSchemaFilter {
+  enum KindFilter {
+    KIND_FILTER_UNSPECIFIED = 0;
+    KIND_FILTER_DEFINITION = 1;
+    KIND_FILTER_CAVEAT = 2;
+    KIND_FILTER_RELATION = 3;
+    KIND_FILTER_PERMISSION = 4;
+  }
+
+  // optional_definition_name_match is a regex that is matched against the definition or caveat name.
+  // If not specified, will be ignored.
+  string optional_definition_name_match = 1;
```

Comment:
Why are these regexes and not strings? I would've expected them to be more like the filters in the Permissions service

Reply:
Purposefully to allow for prefix matching or other forms of matching. For example, you might want to filter to definitions under prefix `someteam/`